File: Dagger2/app/src/main/java/com/example/arnassmicius/dagger2/MainActivity.java
package com.example.arnassmicius.dagger2;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public class MainActivity extends BaseActivity {

    private static final String TAG = "MainActivity";

    private TextView textView1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView1 = findViewById(R.id.tv_text_view_1);

        Log.d(TAG, "onCreate: Before the change Number is : " + prefs.getInt("Number", 0));
        int number = prefs.getInt("Number", 0);
        prefs.edit().putInt("Number", number * 2).apply();
        Log.d(TAG, "onCreate: The value of Number is : " + prefs.getInt("Number", 0));

        textView1.setBackgroundColor(res.getColor(android.R.color.holo_red_dark));
    }
}

File: Retrofit/app/src/main/java/com/example/arnassmicius/retrofit/MainActivity.kt
package com.example.arnassmicius.retrofit

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.widget.Toast

import com.example.arnassmicius.retrofit.api.model.GitHubRepo
import com.example.arnassmicius.retrofit.api.service.GitHubClient

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.rv_list)

        val builder = Retrofit.Builder()
                .baseUrl("https://api.github.com/")
                .addConverterFactory(GsonConverterFactory.create())

        val retrofit = builder.build()

        val client = retrofit.create(GitHubClient::class.java)
        val call = client.reposForUser("ArnasSmicius")

        call.enqueue(object : Callback<List<GitHubRepo>> {
            override fun onResponse(call: Call<List<GitHubRepo>>, response: Response<List<GitHubRepo>>) {
                val repos = response.body()
                Log.d(TAG, "onResponse: listItems: " + repos!!.size)
                val layoutManager = LinearLayoutManager(this@MainActivity, LinearLayoutManager.VERTICAL, false)
                recyclerView.layoutManager = layoutManager
                recyclerView.adapter = GitHubRepoAdapter(this@MainActivity, repos)
            }

            override fun onFailure(call: Call<List<GitHubRepo>>, t: Throwable) {
                Toast.makeText(this@MainActivity, "error :(", Toast.LENGTH_SHORT).show()
            }
        })
    }

    companion object {

        private val TAG = "MainActivity"
    }
}
